### DISCUSSION TITLE
Proposal for Managing Config and Monitoring System via Telegram Bot API

### DISCUSSION THREAD

**Discussion by sentiox** [WGDashboard/WGDashboard#614] [Ideas]
Hi, great that the update is out! My dream is to manage the config through Telegram Bot API using a token, so that it's convenient. I would also like to get information about the processor status, RAM, and other system parameters. When logging into the panel, I would like to see notifications and the IP address, and also manage access according to the admin's ID, for example, through the profile @useridinfobot. =)

**Comment by donaldzou** [comment]
Hi @mizoil, interesting idea! Do you mind to provide some information about the `@useridinfobot` as I don't use Telegram...

and will add the system status to the roadmap in the next release ;)

**Comment by sentiox** [comment]
> Hi @mizoil, interesting idea! Do you mind to provide some information about the `@useridinfobot` as I don't use Telegram...
> 
> and will add the system status to the roadmap in the next release ;)








Hi! Thanks for your interest in my idea. I understand that you don't use Telegram. The idea of using the Telegram Bot API was suggested as a way to easily manage configurations and retrieve system information via a bot. It's just one option, and if Telegram doesn't work for you, here's how it could be organized:

First, a Telegram bot needs to be created with the following functions:

1. Welcome and Main Menu:

- Upon starting the bot, it will send a welcome message:

/start
👋 Hello, [Name]
🤖 Welcome to the control panel bot [link to panel]

------------------------------------------------------------


- Then, the menu will show available actions:

👇 Please choose:
- Button "Create Config" (select WireGuard, VLESS, etc.)
- Button "Server Usage"

------------------------------------------------------------

2. Server Usage Menu:


- The "Server Usage" button will display server information:

💻 Hostname: [NAME]
🚀 WGDashboard Version: v4.1.0
🌐 IPv4: IP
🌐 IPv6: IP
⏳ Server Uptime: 0 Days
📈 Server Load: 0.00, 0.00, 0.00
📋 RAM: 275.68MB/957.43MB
🌐 Clients Online: 0
🔹 Number of TCP connections: 0
🔸 Number of UDP connections: 0
🚦 Traffic: 0GB (↑0GB,↓0GB)
ℹ️ WGDashboard Status: Connected/Disconnected

------------------------------------------------------------

3. Client Management Menu:

- The "All Clients" button will allow the selection of a client, e.g., "wg0."
- The "Select User for wg0 Connection" button will show the selected user, e.g., "Mizoil":

🚨 Enabled: ✅ Yes
🌐 Connection Status: 🔴 Offline or 🟢 Online
💡 Active: ✅ Yes
📅 Expiration Date:
🔼 Outgoing Traffic: ↑ 0 MB/0 GB
🔽 Incoming Traffic: ↓ 0 MB/0 GB
📊 Total: ↑↓ 0 MB/0 GB

------------------------------------------------------------

4. Additional Actions:


- The "Update" button will refresh traffic data.
- The "Reset Traffic" button will reset the traffic in GB and MB.
- The "Change Expiration Date" button will allow users to modify the date or extend the time.
- The "On/Off" button will toggle the VPN connection on or off.
- The "Share" button will allow the user to share the config as a file or a QR code (if supported by the bot), or copy the VLESS key.

5. Login Notifications:

- Every login to the control panel will be recorded and a notification will be sent to the user:

✅ Successful login to the panel.
💻 Hostname: [Name]
👤 Username: [Name]
🌐 IP: [USER IP]
⏰ Time: [DATE TIME]

------------------------------------------------------------

6. Multi-language Support:

- The bot will support language selection based on the user's country, providing convenience for international users.


P.S. This functionality is inspired by the 3x-ui Telegram bot, offering similar features.

**Comment by donaldzou** [comment]
Hi all, I will convert this into a discussion as it might not be the focus of WGDashboard, rather a tool to work with WGDashboard.
